src/main/java/org/prebid/server/bidder/pangle/PangleBidder.java

```diff
@@ -101,13 +104,35 @@ private static int resolveAdType(Imp imp, WrappedImpExtBidder extBidder) {
         throw new PreBidException("not a supported adtype");
     }
 
-    private Imp modifyImp(Imp imp, Integer adType, WrappedImpExtBidder extBidder) {
-        final WrappedImpExtBidder updatedImpExt = extBidder.toBuilder().adType(adType).build();
+    private Imp modifyImp(Imp imp, Integer adType, WrappedImpExtBidder extBidder, ExtImpPangle bidderImpExt) {
+        final NetworkIds modifiedNetworkIds = getNetworkIds(bidderImpExt);
+
+        final WrappedImpExtBidder updatedImpExt = extBidder.toBuilder()
+                .adType(adType)
+                .isPrebid(true)
+                .networkids(modifiedNetworkIds == null ? extBidder.getNetworkids() : modifiedNetworkIds)
+                .build();
+
         return imp.toBuilder()
                 .ext(mapper.mapper().convertValue(updatedImpExt, ObjectNode.class))
                 .build();
     }
 
+    private NetworkIds getNetworkIds(ExtImpPangle bidderImpExt) {
+        if (bidderImpExt != null) {
+            String appid = bidderImpExt.getAppid() == null ? "" : bidderImpExt.getAppid();
+            String placementid = bidderImpExt.getPlacementid() == null ? "" : bidderImpExt.getPlacementid();
```

Comment:
Can be final

Reply:
we don't need to place emptyString if null, it is enough to take it as it is and then instead of str>0, use StringUtils.isNotEmpty()